src/ucp/core/ucp_am.c

```diff
@@ -1334,6 +1373,16 @@ ucp_am_handle_unfinished(ucp_worker_h worker, ucp_recv_desc_t *first_rdesc,
     first_rdesc                   = (ucp_recv_desc_t*)msg - 1;
     first_rdesc->flags            = UCP_RECV_DESC_FLAG_MALLOC;
     first_rdesc->am_malloc_offset = desc_offset;
+    status                        = ucp_am_invoke_cb(worker, am_id, user_hdr,
+                                                     user_hdr_length, msg,
+                                                     total_size, reply_ep,
+                                                     recv_flags);
+    if (ucp_am_rdesc_in_progress(first_rdesc, status)) {
```

Comment:
maybe move ucp_am_rdesc_in_progress inside ucp_am_invoke_cb?
code is similar to line 1242

Reply:
then i'll have to handle `desc==NULL` case inside `ucp_am_rdesc_in_progress`, which is not needed here, but is handled in `ucp_am_handler_common`